### ISSUE TITLE
Add advanced PPO metrics to TensorBoard logging

### ISSUE THREAD

**Issue by rjwalters** [rjwalters/bucket-brigade#62]
## Background

PR #61 successfully added TensorBoard logging to the training pipeline, covering core metrics like policy loss, value loss, entropy, gradient norm, and episode rewards. However, some advanced PPO-specific metrics from issue #13 were not implemented.

## Missing Metrics

These PPO-specific metrics would provide deeper insight into training dynamics:

1. **KL Divergence**: Measures how much the policy changed between updates
   - Useful for detecting policy collapse or instability
   - PPO aims to constrain KL divergence to prevent destructive updates

2. **Clip Fraction**: Percentage of advantages that were clipped by PPO
   - High clip fraction indicates policy is changing rapidly
   - Low clip fraction suggests smaller, safer updates

3. **Explained Variance**: Measures value function quality
   - Ratio of variance in returns explained by value predictions
   - Close to 1.0 = value function accurately predicts returns
   - Close to 0.0 = value function is poor predictor

## Implementation

These metrics need to be calculated in `train_ppo()` at `scripts/train_simple.py:205`:

**KL Divergence:**
```python
with torch.no_grad():
    old_log_probs = dist.log_prob(actions)
# After optimization step:
new_dist = Categorical(logits=policy(observations))
new_log_probs = new_dist.log_prob(actions)
kl_div = (old_log_probs - new_log_probs).mean()
writer.add_scalar("train/kl_divergence", kl_div.item(), global_step)
```

**Clip Fraction:**
```python
ratio = torch.exp(log_probs - old_log_probs)
clipped = torch.clamp(ratio, 1 - clip_eps, 1 + clip_eps)
clip_fraction = (ratio != clipped).float().mean()
writer.add_scalar("train/clip_fraction", clip_fraction.item(), global_step)
```

**Explained Variance:**
```python
with torch.no_grad():
    returns_var = returns.var()
    residual_var = (returns - values).var()
    explained_var = 1 - (residual_var / returns_var)
writer.add_scalar("train/explained_variance", explained_var.item(), global_step)
```

## Success Criteria

- All three metrics logged to TensorBoard during training
- Metrics visible in TensorBoard UI
- Documentation updated to explain what each metric means
- No performance degradation from additional calculations

## Related

- PR #61: Initial TensorBoard integration
- Issue #13: Original request for training visualization
